### PR TITLE
expand content_host.yml to support custom deploy workflows per nick

### PR DESCRIPTION
**Pull request by rplevka**
Partially Addresses: SAT-7681

Please, merge together with the backports:
- 6.10.z: https://github.com/SatelliteQE/robottelo/pull/9336
- 6.9.z:  https://github.com/SatelliteQE/robottelo/pull/9335

This allows us to define custom deploy workflow per individual OS version entry - this is needed as Compose versions are using different naming and thus different deploy workflow.
A sample usage can be seen in the updated `content_host.yml.sample`

### CODE REVIEW

**Comment by rplevka** [comment]
this requires our CI to reflect the changes in configuration. I'll raise an appropriate MR and add a note about the dependency link to this PR

File: pytest_plugins/fixture_markers.py [hunk 1]
```diff
@@ -6,10 +6,14 @@
...
-        rhel_parameters = [
-            dict(workflow=settings.content_host.deploy_workflow, rhel_version=ver)
-            for ver in settings.content_host.rhel_versions
-        ]
+        rhel_parameters = []
+        for ver in settings.content_host.rhel_versions.keys():
+            # prefer nick-specific deploy workflow before using the default one
+            workflow = (
+                settings.content_host.rhel_versions[ver].get('deploy_workflow')
+                or settings.content_host.deploy_workflow
```
Comment (JacobCallahan) [inline pytest_plugins/fixture_markers.py:14]: non-blocking and black will likely adjust formatting
```suggestion
                settings.content_host.rhel_versions[ver].get('deploy_workflow', settings.content_host.deploy_workflow)
```

**Review by JacobCallahan — APPROVED** [review]
ACK. Merge pending testing

**Comment by rplevka** [comment]
Please, do not squash the commits.

File: conf/content_host.yaml.template [hunk 2]
```diff
@@ -18,3 +21,8 @@ content_host:
             RELEASE: 8.3
             MEMORY: 1536 MiB
             CORES: 1
+        RHEL9_compose:
+            RELEASE: 9.0
+            compose: RHEL-9.0.0-20220118.1
```
Comment (jyejare) [inline conf/content_host.yaml.template:26]:
```suggestion
            COMPOSE: RHEL-9.0.0-20220118.1
```

File: pytest_fixtures/core/contenthosts.py [hunk 3]
```diff
@@ -22,8 +22,8 @@ def host_conf(request):
     _rhelver = f"rhel{params.get('rhel_version', settings.content_host.default_rhel_version)}"
     rhel_compose_id = settings.get(f"content_host.hardware.{_rhelver}.compose")
     if rhel_compose_id:
-        conf['rhel_compose_id'] = rhel_compose_id
-    conf['rhel_version'] = settings.content_host.hardware.get(_rhelver).release
+        conf['deploy_rhel_compose_id'] = rhel_compose_id
+    conf['deploy_rhel_version'] = settings.content_host.hardware.get(_rhelver).release
```
Comment (lhellebr) [inline pytest_fixtures/core/contenthosts.py:26]: Why did you rename these two? I don't see them used anywhere. Is it just for consistency?

**Review by lhellebr — APPROVED** [review]
LGTM

File: pytest_fixtures/core/contenthosts.py (hunk 3 above)
Comment (rplevka) [reply pytest_fixtures/core/contenthosts.py:26]: that is an updated workflow argument (the workflow got updated and arguments renamed)

**Review by jyejare — APPROVED** [review]
